lib/vagrant-libvirt/config.rb

```diff
@@ -88,9 +88,10 @@ class Config < Vagrant.plugin('2', :config)
       attr_accessor :machine_virtual_size
       attr_accessor :disk_bus
       attr_accessor :disk_device
+      attr_accessor :disk_driver_opts
       attr_accessor :nic_model_type
       attr_accessor :nested
-      attr_accessor :volume_cache
+      attr_accessor :volume_cache # deprecated, kept for backwards compatibility; use disk_driver
```

Comment:
I'm thinking there must be a standard pattern to make writing to this trigger a deprecated notice.

My guess would be to simply try defining a writer method further down
```
  def volume_cache=(value)
    # emit a deprecation message here
    @volume_cache = value
  end
```

May need to switch the `attr_accessor` to `attr_reader`, not sure whether it will work to override or get an error about having a writer method defined at the same time as `attr_accessor`.

Or is it better done via the validate method? Not sure if that supports the idea of warnings in addition to errors such as the following

https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/config/v2/loader.rb#L111-L137

Reply:
After a bit of tinkering, it seemed that `validate` looked like a sensible place to put it; it generally gets called once, and the vagrant environment is available to properly show a pretty colourised warning message. It looks like the `validate` return value is only for errors, but using the ui.warn method seems to work nicely.

Reply:
That looks like it'll work nicely enough